components/constellation/constellation.rs

```diff
@@ -1596,13 +1600,9 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
                 Some(source_id)
             }
             None => {
-                // Make sure no pending page would be overridden.
-                for frame_change in &self.pending_frames {
-                    if frame_change.old_pipeline_id == Some(source_id) {
-                        // id that sent load msg is being changed already; abort
-                        return None;
-                    }
-                }
+                // Cancel any in progress loads for this frame
+                // https://html.spec.whatwg.org/multipage/#navigate step 6
+                self.clear_pending_loads_for_frame(frame_id);
```

Comment:
If the navigation is occurring on a nested frame, it is rerouted to the script thread which then will send the constellation a `ScriptLoadedURLInIFrame` message. The pending loads are cleared when that is handled in the constellation.